File: backend/src/main/java/net/zorphy/backend/site/jolly/dto/game/GameStats.java
package net.zorphy.backend.site.jolly.dto.game;

import net.zorphy.backend.main.game.dto.stats.GameSpecificStats;
import net.zorphy.backend.main.game.dto.stats.metrics.GameStatsMetrics;
import net.zorphy.backend.main.game.dto.stats.streaks.GameStatsStreak;

import java.time.Duration;


public record GameStats(
        int roundsPlayed,
        double roundWinRate,
        GameStatsMetrics<Double> roundsMetrics,
        GameStatsMetrics<Double> roundScoreMetrics,
        GameStatsMetrics<Duration> roundDurationMetrics,
        double outInOneRate,
        double closedRate,
        GameStatsStreak maxOutInOneStreak,
        GameStatsStreak maxClosedStreak,
        double jokersPerRound
) implements GameSpecificStats {
}

File: frontend/src/app/sites/jolly/dto/game/GameStats.ts
import {GameStatsMetrics} from "../../../../main/games/dto/stats/metrics/GameStatsMetrics";
import {GameStatsStreak} from "../../../../main/games/dto/stats/streaks/GameStatsStreak";

export interface GameStats {
    roundsPlayed: number,
    roundWinRate: number,
    roundScoreMetrics: GameStatsMetrics<number>,
    roundDurationMetrics: GameStatsMetrics<string>,
    outInOneRate: number,
    closedRate: number,
    maxOutInOneStreak: GameStatsStreak,
    maxClosedStreak: GameStatsStreak,
}